dom/src/main/java/org/apache/james/mime4j/field/address/DefaultAddressParser.java

```diff
@@ -85,8 +83,12 @@ public AddressList parseAddressList(CharSequence text) throws ParseException {
      *             address.
      */
     public Mailbox parseMailbox(CharSequence text, DecodeMonitor monitor) throws ParseException {
-        AddressListParser parser = new AddressListParser(InputStreams.create(text, Charsets.UTF_8));
-        return Builder.getInstance().buildMailbox(parser.parseMailbox(), monitor);
+        return Builder.getInstance().buildMailbox(createParser(text).parseMailbox(), monitor);
+    }
+
+    private AddressListParser createParser(CharSequence text) {
```

Comment:
where are the tests ?

Reply:
Perhaps it would be not easy to covert this, as one need to set `file.encoding` system property before using `DefaultAddressParser.DEFAULT`. In other words `DefaultAddressParser.DEFAULT` cannot be used and `DefaultAddressParser` instance should be created using the constructor.

Reply:
Unfortunately, yes, the file.encoding property has to be specified as the JVM starts up; by the time your main method is entered, the character encoding used by String.getBytes() and the default constructors of InputStreamReader and OutputStreamWriter has been permanently cached.

We can not override this in the tests....

Reply:
Depending on how deep you can go, one can nullify `java.nio.charset.Charset#defaultCharset` field using reflection:
`org.apache.commons.lang3.reflect.FieldUtils.writeStaticField(Charset.class, "defaultCharset", null, true);`
I haven't tried this code so I am not sure it works :) Should work.